app/Http/Controllers/DiceController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Validator;
+use Illuminate\Http\Request;
+
+class DiceController extends Controller
+{
+    public function roll(Request $request)
+    {
+        /*
+         * /\d+d\d+/ matches for a single command (1d10, 5d5, 10d20...)
+         * $diceRegex matches for either a roll or a comma separated list.
+         * Note that this doesn't capture. It is just used for validation.
+         */
+        $diceRegex = "/^\d+d\d+(,\d+d\d+)*$/i";
+
+        $validator = Validator::make($request->all(), [
+            "q" => ["required", "regex:$diceRegex"],
+        ], [
+            /* Provide more friendly messages when q is not valid. */
+            "q.required" => "You must provide a roll set.",
+            "q.regex" => "You provided an invalid roll set.",
+        ]);
+        if ($validator->fails()) {
+            return response()->json([
+                "errors" => $validator->errors()->all(),
+            ], 422);
+        }
+
+        $rollCommand = strtolower($request->get("q"));
+        return response()->json($this->processRoll($rollCommand));
+    }
+
+    /**
+     * Actually processes a roll set combination.
+     *
+     * This function assumes that the roll set has already be validated and
+     * it is valid or at least not malformed. The combination is a string
+     * of comma separated rolls, each roll having the form xDy, where x and y
+     * being natural numbers with different meanings. We will ignore the
+     * dice semantics here and accept any kind of dices for the moment, so
+     * rolls such as 10d25 or 13D26 are totally valid here.
+     *
+     * @param string $rollSet the list of rolls to yield.
+     * @return array processed list of rolls.
+     */
+    private function processRoll(string $rollSet)
+    {
+        /* Accumulator object to store partial results of rolls. */
```

Comment:
Solo una duda aquí, ¿es mejor a la hora de trabajar con JS esta respuesta que algo de este estilo?
`{"dice":2,"rolls":[{"size":4,"outcome":2},{"size":8,"outcome":8}],"sum":10}`
¿Cómo lo ves más claro?

Cuidado con la palabra "dado" en inglés que es un poco rara. "Die" es dado en singular y "dice" en plural. "Dices" no existe o al menos yo no lo he visto en ningún libro de rol.

Reply:
He considerado que sería más compacto que estar todo el rato repitiendo las keys, aunque me es indiferente, la verdad. En un caso sacrificas tamaño del JSON devuelto, en el otro iterar indizadamente para sacar ambas propiedades.

> Cuidado con la palabra "dado" en inglés que es un poco rara. "Die" es dado en singular y "dice" en plural. "Dices" no existe o al menos yo no lo he visto en ningún libro de rol.

Va, a esto es a lo que me refería cuando en el texto de la PR decía que los nombres son mejorables. No sabía si die era un typo o no, y tampoco me he ido al diccionario a ver si estaba yo equivocado y "dices" no existe. Aunque es verdad que yo siempre he escuchado "dice", incluso cuando se refieren a un único dado.

Reply:
Hehe, https://grammarist.com/usage/dice-die/:

> Because the irregular plural is more common than the singular, many writers forget to use the singular at all

Lo anoto para corregir mañana en un momento, aunque sólo sea por usar adecuadamente la lengua de Shakspeare.

Reply:
En este caso prefiero que la salida sea como te indico. Es que el tema de andar con el mismo índice para recoger el valor de dos arrays me mola poco o nada. Manías mías.

Reply:
Actualizado.